llvmlite/binding/newpassmangers.py

```diff
@@ -0,0 +1,305 @@
+        super().__init__(ptr)
+
+    def run(self, module, pb):
+        ffi.lib.LLVMPY_RunNewModulePassManager(self, pb, module)
+
+    def add_verifier(self):
+        ffi.lib.LLVMPY_AddVerifierPass(self)
+
+    def add_aa_eval_pass(self):
+        ffi.lib.LLVMPY_AddAAEvalPass_module(self)
+
+    def add_simplify_cfg_pass(self):
+        ffi.lib.LLVMPY_AddSimplifyCFGPass_module(self)
+
+    def add_loop_unroll_pass(self):
+        ffi.lib.LLVMPY_AddLoopUnrollPass_module(self)
+
+    def add_loop_rotate_pass(self):
+        ffi.lib.LLVMPY_AddLoopRotatePass_module(self)
+
+    def add_instruction_combine_pass(self):
+        ffi.lib.LLVMPY_AddInstructionCombinePass_module(self)
+
+    def add_jump_threading_pass(self, threshold=-1):
+        ffi.lib.LLVMPY_AddJumpThreadingPass_module(self, threshold)
+
+    def _dispose(self):
+        ffi.lib.LLVMPY_DisposeNewModulePassManger(self)
+
+
+class FunctionPassManager(ffi.ObjectRef):
+
+    def __init__(self, ptr=None):
+        if ptr is None:
+            ptr = ffi.lib.LLVMPY_CreateNewFunctionPassManager()
+        super().__init__(ptr)
+
+    def run(self, fun, pb):
+        ffi.lib.LLVMPY_RunNewFunctionPassManager(self, pb, fun)
+
+    def add_aa_eval_pass(self):
+        ffi.lib.LLVMPY_AddAAEvalPass_function(self)
+
+    def add_simplify_cfg_pass(self):
+        ffi.lib.LLVMPY_AddSimplifyCFGPass_function(self)
+
+    def add_loop_unroll_pass(self):
+        ffi.lib.LLVMPY_AddLoopUnrollPass_function(self)
+
+    def add_loop_rotate_pass(self):
+        ffi.lib.LLVMPY_AddLoopRotatePass_function(self)
+
+    def add_instruction_combine_pass(self):
+        ffi.lib.LLVMPY_AddInstructionCombinePass_function(self)
+
+    def add_jump_threading_pass(self, threshold=-1):
+        ffi.lib.LLVMPY_AddJumpThreadingPass_function(self, threshold)
+
+    def _dispose(self):
+        ffi.lib.LLVMPY_DisposeNewFunctionPassManger(self)
+
+
+class PipelineTuningOptions(ffi.ObjectRef):
+
+    def __init__(self, speed_level=2, size_level=0):
+        self._speed_level = None
+        self._size_level = None
+        self.speed_level = speed_level
+        self.size_level = size_level
+        super().__init__(ffi.lib.LLVMPY_CreatePipelineTuningOptions())
+
+    @property
+    def speed_level(self):
+        return self._speed_level
+
+    @speed_level.setter
+    def speed_level(self, value):
+        if not 0 <= value <= 3:
+            raise ValueError(
+                "Optimization level for speed should be 0, 1, 2, or 3")
+        self._speed_level = value
+
+    @property
+    def size_level(self):
+        return self._size_level
+
+    @size_level.setter
+    def size_level(self, value):
+        if not 0 <= value <= 2:
+            raise ValueError("Optimization level for size should be 0, 1, or 2")
+        if value != 0 and self.speed_level != 2:
+            raise ValueError(
+                "Optimize for size should be encoded with speedup level == 2")
```

Comment:
Maybe just a slight clarification of the message:
```suggestion
                "Optimization for size should be encoded with speed level == 2")
```